packages/keyring-eth-simple/src/simple-keyring-v2.ts

```diff
@@ -0,0 +1,226 @@
+      if (options.type !== 'private-key:import') {
+        throw new Error(
+          `Unsupported account creation type for SimpleKeyring: ${options.type}`,
+        );
+      }
+
+      // Validate account type
+      if (options.accountType !== EthAccountType.Eoa) {
+        throw new Error(
+          `Unsupported account type for SimpleKeyring: ${options.accountType}. Only '${EthAccountType.Eoa}' is supported.`,
+        );
+      }
+
+      const { encoding, privateKey } = options;
+
+      // Validate encoding
+      if (encoding !== PrivateKeyEncoding.Hexadecimal) {
+        throw new Error(
+          `Unsupported encoding for SimpleKeyring: ${encoding}. Only '${PrivateKeyEncoding.Hexadecimal}' is supported.`,
+        );
+      }
+
+      // Get current accounts to preserve them
+      const currentAccounts = await this.inner.serialize();
+
+      // Import the new private key by deserializing with all accounts
+      await this.inner.deserialize([...currentAccounts, privateKey]);
+
+      // Get the address of the newly added account
+      const allAddresses = await this.inner.getAccounts();
+      const newAddress = allAddresses[allAddresses.length - 1];
+
+      if (!newAddress) {
+        throw new Error('Failed to import private key');
+      }
+
+      // Create and return the new KeyringAccount
+      const newAccount = this.#createKeyringAccount(newAddress);
+      return [newAccount];
+    });
+  }
+
+  /**
+   * Delete an account from the keyring.
+   *
+   * @param accountId - The account ID to delete.
+   */
+  async deleteAccount(accountId: AccountId): Promise<void> {
+    await this.#lock.runExclusive(async () => {
+      const account = await this.getAccount(accountId);
+
+      // Remove from the legacy keyring
+      this.inner.removeAccount(account.address);
+
+      // Remove from the registry
+      this.registry.delete(accountId);
+    });
+  }
+
+  /**
+   * Export the private key for an account in hexadecimal format.
+   *
+   * @param accountId - The ID of the account to export.
+   * @param options - Export options (only hexadecimal encoding is supported).
+   * @returns The exported account with private key.
+   */
+  async exportAccount(
+    accountId: AccountId,
+    options?: ExportAccountOptions,
+  ): Promise<ExportedAccount> {
+    const account = await this.getAccount(accountId);
+
+    const requestedEncoding =
+      options?.encoding ?? PrivateKeyEncoding.Hexadecimal;
+
+    if (requestedEncoding !== PrivateKeyEncoding.Hexadecimal) {
+      throw new Error(
+        `Unsupported encoding for SimpleKeyring: ${requestedEncoding}. Only '${PrivateKeyEncoding.Hexadecimal}' is supported.`,
+      );
+    }
+
+    const privateKeyHex = await this.inner.exportAccount(
+      this.toHexAddress(account.address),
+    );
+
+    const exported: ExportedAccount = {
+      type: 'private-key',
+      privateKey: `0x${privateKeyHex}`,
+      encoding: PrivateKeyEncoding.Hexadecimal,
+    };
```

Comment:
WDYT about re-using a similar logic than for: https://github.com/MetaMask/accounts/blob/main/packages/keyring-eth-hd/src/hd-keyring-v2.ts#L267-L277

Relying on `add0x` mostly?